src/condor_shadow.V6.1/remoteresource.cpp

```diff
@@ -904,6 +914,63 @@ RemoteResource::setStarterInfo( ClassAd* ad )
 				 ATTR_HAS_RECONNECT );
 	}
 
+
+	//
+	// If the starter is too old for common file transfer, fall back on
+	// per-proc file transfer.
+	//
+	CondorVersionInfo cvi( starter_version.c_str() );
+	// `#define CFT_VERSION 2` went in with HTCONDOR-3168, which was first
+	// actually released as part of 25.2.FIXME.
+	//
+	// CFT_VERSION = 1 starters (set in HTCONDOR-3051, and released as 24.9.0)
+	// can successfully do common file transfer if and only if there were no
+	// catalogs specified and the job ad has the test syntax from HTC25.  As
+	// a result, those will be treated as needing the fall-back as well.
+	//
+	bool disallowed = param_boolean("FORBID_COMMON_FILE_TRANSFER", false);
+	if( disallowed || (! cvi.built_since_version( 25, 2, 0 )) ) {
+		auto common_file_catalogs = computeCommonInputFileCatalogs( jobAd, shadow );
+		if(! common_file_catalogs) {
+			dprintf( D_ERROR, "Failed to construct unique name for catalog, can't run job!\n" );
+
+			// We don't have a mechanism to inform the submitter of internal
+			// errors like this, so for now we're stuck putting the job on hold.
+			shadow->holdJob( "Internal error: failed to construct unique name for catalog.",
+				CONDOR_HOLD_CODE::JobNotStarted, 4
+			);
+
+			return;
+		}
+
+		int required_version = 2;
+		if(! computeCommonInputFiles( jobAd, shadow, *common_file_catalogs, required_version )) {
+			dprintf( D_ERROR, "Failed to construct unique name for catalog, can't run job!\n" );
+
+			// We don't have a mechanism to inform the submitter of internal
+			// errors like this, so for now we're stuck putting the job on hold.
+			shadow->holdJob( "Internal error: failed to construct unique name for catalog.",
+				CONDOR_HOLD_CODE::JobNotStarted, 4
+			);
+
+			return;
+		}
+
+		if( common_file_catalogs->empty() ) {
+			return;
+		}
+
+		std::string whole_list;
+		for( const auto & [cifName, commonInputFiles] : *common_file_catalogs ) {
+			dprintf( D_ZKM, "%s = %s\n", cifName.c_str(), commonInputFiles.c_str() );
+
+			std::set< std::string > pathsAlreadyPreserved;
```

Comment:
Unused variable? I assume this was from when you were using the old `addInputFile()` before rename (i.e. `addInputFileEx()`

Reply:
Yes, and I'm a little annoyed that the compiler didn't complain.